app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -126,4 +171,77 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
 
     private fun getMoneyButtonIds() = arrayOf(btn_tip, btn_currency, btn_taxes)
     private fun getButtonIds() = arrayOf(btn_decimal, btn_0, btn_1, btn_2, btn_3, btn_4, btn_5, btn_6, btn_7, btn_8, btn_9)
+
+
+    private inner class JsonTask : AsyncTask<String, String, String>() {
+
+        override fun onPreExecute() {
+            super.onPreExecute()
+
+            pd = ProgressDialog(this@MoneyActivity)
+            pd?.setMessage("Fetching currency conversion rates.")
+            pd?.setCancelable(false)
+            pd?.show()
+        }
+
+        override fun doInBackground(vararg params: String): String {
+
+
+            var connection: HttpURLConnection? = null
+            var reader: BufferedReader? = null
+
+            try {
+                val url = URL(params[0])
+                connection = url.openConnection() as HttpURLConnection
+                connection.connect()
+
+
+                val stream = connection.inputStream
+
+                reader = BufferedReader(InputStreamReader(stream))
+
+                val buffer = StringBuffer()
+                var line: String? = null
+
+                while (true) {
+                    line = reader.readLine()
+                    if (line == null) break
+                    buffer.append(line + "\n")
+                    Log.d("Response: ", "> $line")   //here u ll get whole response...... :-)
+
+                }
+
+                return buffer.toString()
+
+
+            } catch (e: MalformedURLException) {
+                e.printStackTrace()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            } finally {
+                if (connection != null) {
+                    connection.disconnect()
+                }
+                try {
+                    if (reader != null) {
+                        reader.close()
+                    }
+                } catch (e: IOException) {
+                    e.printStackTrace()
+                }
+
+            }
+            return ""
```

Comment:
In the return type of this method header instead of just String add in String? in order to get rid of this return "" statement

Reply:
A return statement is still necessary in a function with a block body. I can add String? as the return type and return null instead.